### PR TITLE
Accept other ios_banner types

### PR DESCRIPTION
**Pull request by codemonkeyjim**
##### SUMMARY
Update `ios_banner` to accept all banner types in Cisco IOS 15M. Specifically, this change adds `exec`, `incoming`, and `slip-ppp` to the list of accepted banner types.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ios_banner

##### ANSIBLE VERSION
```
ansible 2.5.0 (add_ios_banner_choices 5839342daf) last updated 2017/12/15 14:17:48 (GMT -700)
  config file = /Users/pflegge/.ansible.cfg
  configured module search path = [u'/Users/pflegge/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/pflegge/Projects/ansible/ansible/lib/ansible
  executable location = /Users/pflegge/Projects/ansible/ansible/bin/ansible
  python version = 2.7.14 (default, Dec 12 2017, 14:47:14) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### ADDITIONAL INFORMATION
[Banner command in *Configuration Fundamentals Configuration Guide, Cisco IOS Release 15M&T*](https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/fundamentals/configuration/15mt/fundamentals-15-mt-book/cf-connections.html#GUID-D176A70A-EA99-4BF9-BBA6-AED5383D5729)

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_banner.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
         configured on the remote device.
```
Comment (gundalow) [inline lib/ansible/modules/network/ios/ios_banner.py]: Please state that "in Ansible 2.4 and earlier only I(login) and I(motd) were supported."

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
Unit tests are failing on **Python 2.6**


```
self = <units.modules.network.ios.test_ios_banner.TestIosBannerModule testMethod=test_ios_banner_create>

    def test_ios_banner_create(self):
        for banner_type in ('login', 'motd', 'exec', 'incoming', 'slip-ppp'):
            set_module_args(dict(banner=banner_type, text='test\nbanner\nstring'))
>           commands = ['banner {} @\ntest\nbanner\nstring\n@'.format(banner_type)]
E           ValueError: zero length field name in format

test/units/modules/network/ios/test_ios_banner.py:51: ValueError
```

**Comment by codemonkeyjim** [comment]
Requested changes made to docs and to get 2.6 tests passing.

3.7 tests are failing for me because I can't get dependencies to install. Do I need to get 3.7 passing for this PR?

**Comment by gundalow** [comment]
Python 2.6, 3.5 and 3.5 must pass. Looks like they are now.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -11,8 +11,9 @@ Also test if the latest release, and master branch are affected too.
 
 ##### COMPONENT NAME
 <!---
-Name of the module/plugin/task/feature
-Do not include extra details here, e.g. "vyos_command" not "the network module vyos_command" or the full path-->
+Name of the module, plugin, task or feature
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE.md]: Not sure why this is here

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ the change does.
  - Docs Pull Request
 
 ##### COMPONENT NAME
-<!--- Name of the module/plugin/module/task -->
+<!--- Name of the module, plugin, module or task -->
```
Comment (gundalow) [inline .github/PULL_REQUEST_TEMPLATE.md]: Unrelated change

**Comment by gundalow** [comment]
http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html details how to cleanly rebase

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (codemonkeyjim) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'm sorry about that; this is my first-ever PR. I think I fixed this following the rebase instructions, but I also thought I'd done this right the first time. 😕

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into devel will be released in 2.5.0.

Thanks!
